### PR TITLE
enhancement: add new checker 'consider-using-dict-comprehension'

### PR DESCRIPTION
**Pull request by pheanex**
As discussed in #2194 this adds the new checker for using old style dict initializations with list comprehensions instead of dict-comprehensions.
How is this for a first version?
I think it could use some more tests, but could not think of any more at the moment :smile: 
Anyone see any edge-cases?

### CODE REVIEW

File: pylint/checkers/refactoring.py [hunk 1]
```diff
@@ -434,9 +439,19 @@ def _check_exception_inherit_from_stopiteration(exc):
         stopiteration_qname = '{}.StopIteration'.format(utils.EXCEPTIONS_MODULE)
         return any(_class.qname() == stopiteration_qname for _class in exc.mro())
 
-    @utils.check_messages('stop-iteration-return')
+    def _check_consider_using_dict_comprehension(self, node):
+        if (hasattr(node, 'func') and
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: The node always has `func`, so the following should do. Note that its preferable to use `isinstance` against the correct type of node you expect rather than `hasattr`.
```
if isinstance(node.func, astroid.Name) and node.func.name == 'dict'
```

File: pylint/checkers/refactoring.py [hunk 2]
```diff
@@ -434,9 +439,19 @@ def _check_exception_inherit_from_stopiteration(exc):
...
+                hasattr(node.func, 'name') and
+                node.func.name == 'dict' and
+                len(node.args) and
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: `and node.args` should be enough.

File: pylint/checkers/refactoring.py [hunk 3]
```diff
@@ -159,7 +159,12 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                   'chained-comparison',
                   'This message is emitted when pylint encounters boolean operation like'
                   '"a < b and b < c", suggesting instead to refactor it to "a < b < c"',
-                  ),
+                 ),
+        'R1717': ('Consider using new dict initialization syntax',
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: `Consider using a dictionary comprehension` sounds better.

File: pylint/checkers/refactoring.py [hunk 4]
```diff
@@ -159,7 +159,12 @@ class RefactoringChecker(checkers.BaseTokenChecker):
...
+                  'consider-using-dict-comprehension',
+                  'Although there is nothing syntactically wrong with this code, '
+                  'it is hard to read and can be simplified to a dict comprehension',
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: You can also mention the fact that it is faster since you don't create another transient list.

**Comment by PCManticore** [comment]
Thank you @pheanex
